src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -2560,7 +2546,7 @@ static void read_action_locked(grpc_exec_ctx *exec_ctx, void *tp,
     grpc_endpoint_read(exec_ctx, t->ep, &t->read_buffer,
                        &t->read_action_locked);
     grpc_chttp2_act_on_flowctl_action(
-        exec_ctx, grpc_chttp2_flowctl_get_bdp_action(&t->flow_control), t,
+        exec_ctx, grpc_chttp2_flowctl_get_action(&t->flow_control, NULL), t,
```

Comment:
Merging these two functions into one causes the bdp_est codepath to be taken about 3X more often. I would suggest keeping them distinct, or adding a flag like check_for_bdp_updates. That is why the count for gpr_now is spiking in in bm_diff.

Reply:
#11866 will sort out timing stuff nicely in a future rev